### PR TITLE
feat(releases): Add POST endpoint for thresholds

### PR DESCRIPTION
**Pull request by EricHasegawa**
Adds basic `POST` endpoint to create new release thresholds

### CODE REVIEW

File: src/sentry/api/endpoints/release_threshold.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.environment import EnvironmentField
+from sentry.api.serializers.rest_framework.project import ProjectField
+from sentry.models import Project
+from sentry.models.release_threshold.constants import (
+    THRESHOLD_TYPE_STR_TO_INT,
+    TRIGGER_TYPE_STRING_TO_INT,
+)
+from sentry.models.release_threshold.releasethreshold import (
```
Comment (nhsiehgit) [inline src/sentry/api/endpoints/release_threshold.py:17]: Ah - if you're adding the folder to the models, you can just leave this one as the `__init__.py` file maybe?

Comment (nhsiehgit) [reply src/sentry/api/endpoints/release_threshold.py:17]: There aren't currently many model folders though (🤷 )

So maybe we move the constants to the api/endpoints/release_thresholds?

File: src/sentry/api/endpoints/release_threshold.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+class ReleaseThresholdSerializer(serializers.Serializer):
+    threshold_type = serializers.ChoiceField(choices=ReleaseThresholdType.as_str_choices())
+    trigger_type = serializers.ChoiceField(choices=TriggerType.as_str_choices())
+    value = serializers.IntegerField()
+    window_in_seconds = serializers.IntegerField()
+    project = ProjectField()
+    environment = EnvironmentField(required=False, allow_null=True)
+
+    def validate_threshold_type(self, threshold_type):
+        return THRESHOLD_TYPE_STR_TO_INT[threshold_type]
```
Comment (nhsiehgit) [inline src/sentry/api/endpoints/release_threshold.py:33]: Is this just to validate whether the passed threshold type is a valid defined type?

If this returns falsey (eg. 0) will this fail?
And do we need to actually have a mapping of the string -> int for this?

File: src/sentry/api/endpoints/release_threshold.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+class ReleaseThresholdSerializer(serializers.Serializer):
+    threshold_type = serializers.ChoiceField(choices=ReleaseThresholdType.as_str_choices())
+    trigger_type = serializers.ChoiceField(choices=TriggerType.as_str_choices())
+    value = serializers.IntegerField()
+    window_in_seconds = serializers.IntegerField()
+    project = ProjectField()
+    environment = EnvironmentField(required=False, allow_null=True)
+
+    def validate_threshold_type(self, threshold_type):
```
Comment (nhsiehgit) [inline src/sentry/api/endpoints/release_threshold.py:32]: nit: add typings here?

File: src/sentry/api/serializers/models/release_threshold.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.models.release_threshold.constants import (
```
Comment (nhsiehgit) [inline src/sentry/api/serializers/models/release_threshold.py:2]: Maybe `utils` is more appropriate 🤷 
I dunno haha. 

Just a thought - i think constants is probably fine tho

File: src/sentry/api/urls.py [hunk 5]
```diff
@@ -2100,6 +2101,11 @@
         ProjectReleasesEndpoint.as_view(),
         name="sentry-api-0-project-releases",
     ),
+    re_path(
+        r"^(?P<organization_slug>[^\/]+)/(?P<project_slug>[^\/]+)/releases/thresholds/$",
+        ReleaseThresholdEndpoint.as_view(),
```
Comment (nhsiehgit) [inline src/sentry/api/urls.py:2106]: I imagine one day we'll have a `threshold details` api as well.

Maybe we call this class the `ReleaseThresholdIndex` instead? (do they all need to be called `...Endpoint`?)
Index post being the generic "create a trheshold", or maybe even can update to pass a list of thresholds to create
Details being to modify a specific threshold (PUT/DELETE/GET)

File: tests/sentry/api/endpoints/test_release_threshold.py [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+        assert response.status_code == 400
+        assert response.data["trigger_type"][0].code == "invalid_choice"
+
+    def test_invalid_project(self):
+        url_with_invalid_project = reverse(
+            "sentry-api-0-project-release-thresholds",
+            kwargs={
+                "organization_slug": self.organization.slug,
+                "project_slug": "Why did it have to be snakes?",
```
Comment (nhsiehgit) [inline tests/sentry/api/endpoints/test_release_threshold.py:75]: 😂

**Review by nhsiehgit — COMMENTED** [review]
nice nice nice

a few questions/comments

File: src/sentry/api/endpoints/release_threshold.py (hunk 1 above)
Comment (EricHasegawa) [reply src/sentry/api/endpoints/release_threshold.py:17]: Synced in person re. circular imports

File: src/sentry/api/endpoints/release_threshold.py (hunk 2 above)
Comment (EricHasegawa) [reply src/sentry/api/endpoints/release_threshold.py:33]: This actually converts it to an `int` for us so when we use it in the creation endpoint it gets inserted properly. We'll need the mapping for the conversion and I figured this was a good place for it - lmk if you disagree

Comment (nhsiehgit) [reply src/sentry/api/endpoints/release_threshold.py:31]: Oh mmm

I guess i meant the `validate_threshold_type` method.
Does this actually modify the serializer response? or just validates it?

I guess i can just look at how its used in the super 😅

Comment (nhsiehgit) [reply src/sentry/api/endpoints/release_threshold.py:31]: I see.
So this _does_ transform the form values
Ok - so we probably wawnt to throw a validation error if the key doesn't exist?

File: src/sentry/api/urls.py (hunk 5 above)
Comment (nhsiehgit) [reply src/sentry/api/urls.py:2106]: Ah that's fair 👍

**Review by nhsiehgit — APPROVED** [review]
sick sick sick 👍 

quick comment on the validatior response, but non blocking

**Pull request merged**